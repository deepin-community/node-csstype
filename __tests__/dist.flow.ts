import { spawnSync } from 'child_process';
import * as path from 'path';

const flow: string = require('flow-bin');

test('it detects errors', () => {
  const { stdout: output } = spawnSync(
    flow,
    ['check', path.resolve(__dirname, '__fixtures__/typecheck.js'), '--json'],
    {
      stdio: 'pipe',
      encoding: 'utf8',
    },
  );

  const { errors }: IFlowOutput = JSON.parse(output);

  expect(errors.length).toBe(4);
  for (const error of errors) {
    expect(message(error)).toMatchSnapshot();
  }
});

function message(error: IError) {
  const {
    loc: {
      start: { line, column },
    },
    descr,
  } = error.message[0];

  return `${line}:${column} - ${descr}`;
}

interface IExtra {
  message: IMessage[];
}

interface IPosition {
  line: number;
  column: number;
  offset: number;
}

interface ILocation {
  source: string;
  type: string;
  start: IPosition;
  end: IPosition;
}

interface IMessage {
  context: string;
  descr: string;
  type: string;
  loc: ILocation;
  path: string;
  line: number;
  endline: number;
  start: number;
  end: number;
}

interface IError {
  kind: string;
  level: string;
  suppressions: any[];
  extra: IExtra[];
  message: IMessage[];
}

interface IFlowOutput {
  flowVersion: string;
  jsonVersion: string;
  errors: IError[];
  passed: boolean;
}
