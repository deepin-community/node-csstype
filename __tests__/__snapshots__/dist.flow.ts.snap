// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it detects errors 1`] = `"47:43 - Cannot assign object literal to \`unknownProperty\` because property \`unknownProperty\` is missing in \`Properties\` [1] but exists in object literal [2]. [prop-missing]"`;

exports[`it detects errors 2`] = `"54:10 - Cannot assign object literal to \`cssWithDisallowedValue\` because in property \`order\`: [incompatible-type] Either string [1] is incompatible with literal union [2]. Or string [1] is incompatible with number [3]."`;

exports[`it detects errors 3`] = `"60:11 - Cannot assign object literal to \`cssWithDisallowedFallbackValues\` because in property \`bottom\`: [incompatible-type] Either array literal [1] is incompatible with literal union [2]. Or array literal [1] is incompatible with string [3]. Or array literal [1] is incompatible with number literal \`0\` [4]. Or array literal [1] is incompatible with string literal \`auto\` [5]. Or array literal [1] is incompatible with string [6]."`;

exports[`it detects errors 4`] = `"61:10 - Cannot assign object literal to \`cssWithDisallowedFallbackValues\` because in property \`order\`: [incompatible-type] Either array literal [1] is incompatible with literal union [2]. Or array literal [1] is incompatible with number [3]."`;
